tests/test_crs.py

```diff
@@ -0,0 +1,289 @@
+"""
+==============
+test_crs.py
+==============
+
+Test the net2cog utilites functionality.
+"""
+
+import pathlib
+import os
+
+import pytest
+import numpy as np
+import xarray as xr
+from rasterio.crs import CRS
+
+from net2cog.netcdf_convert import (
+    Net2CogError,
+    get_crs_from_grid_mapping,
+)
+from net2cog.utilities import resolve_relative_path
+
+
+def test_crs_nested_path_same_group(temp_dir, logger, spl2smp_nested_file):
+    """Test CRS variable with nested path same group"""
+
+    test_crs = CRS.from_epsg(6933)
+    test_file = pathlib.Path(temp_dir, spl2smp_nested_file)
+    netcdf_file = os.path.abspath(test_file)
+    input_datatree = xr.open_datatree(netcdf_file)
+
+    # Process test file:
+    output_crs = get_crs_from_grid_mapping(
+        input_datatree, "Soil_Moisture_Retrieval_Data/vegetation_water_content", logger
+    )
+
+    assert output_crs.to_epsg() == test_crs
+    assert output_crs.is_projected
+
+
+def test_crs_nested_path_root_group(temp_dir, logger, nested_file):
+    """Test CRS variable with nested path at root group"""
+
+    test_crs = CRS.from_epsg(6933)
+    test_file = pathlib.Path(temp_dir, nested_file)
+    netcdf_file = os.path.abspath(test_file)
+    input_datatree = xr.open_datatree(netcdf_file)
+
+    # Process test file:
+    output_crs = get_crs_from_grid_mapping(input_datatree, "NEE/nee_mean", logger)
+
+    assert output_crs.to_epsg() == test_crs
+    assert output_crs.is_projected
+
+
+def test_crs_multiple_variable_selection_no_grid_mapping(temp_dir, smap_file, logger):
+    """Test the default EPSG:4326 CRS variable's handling of the absence of grid
+    mapping attributes.
+
+    """
+
+    test_file = pathlib.Path(temp_dir, smap_file)
+
+    netcdf_file = os.path.abspath(test_file)
+    input_datatree = xr.open_datatree(netcdf_file)
+
+    # Process test file:
+    output_crs = get_crs_from_grid_mapping(input_datatree, "sss_smap", logger)
+
+    result_crs = output_crs.to_epsg()
+
+    assert result_crs == 4326
+    assert output_crs.is_geographic
+
+
+@pytest.fixture(scope="class")
```

Comment:
Nice fixture. A couple of suggestions:

* This should probably go in `conftest.py`, which is the standard place to put fixtures. (`pytest` can automatically discover fixtures in a `conftest.py` file, so you don't need to import then fixtures, wonky though that feels)
* I think you probably can drop the `scope` and let it be unique per test. Right now, I'm not sure if the `scope="class"` is buying you anything because the tests aren't in a class, so it's essentially the same as having them scoped to each test function. That level of scope (a unique instance per test) feels right to me, just in case the `DataTree` gets mutated in some way, as it is just stored in-memory.